filelog_unix.go

```diff
@@ -2,4 +2,69 @@
 
 package log4go
 
+import (
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+)
+
 const FILELOG_ARCHIVE_REGEX = `\.[0-9]{4}-[0-9]{2}-[0-9]{2}(\.[0-9]{4})?(\.gz)?$`
+
+func (w *FileLogWriter) compressFile(filename string) {
+	basename := filepath.Base(filename)
+	dirname := filepath.Dir(filename)
+
+	plainFile, err := os.Open(filename)
+	if err != nil {
+		fmt.Fprintf(w.errorWriter, "FileLogWriter(%q): Couldn't open logfile to begin compression: %s\n", w.filename, err)
+		return
+	}
+
+	compressedFilename := filepath.Join(dirname, basename+".gz")
+	compressedFile, err := os.Create(compressedFilename + ".inprogress")
```

Comment:
Two benefits: 1) its easier to see when a large compressed file is finished writing (many programs like Chrome do this too), 2) its immediately obvious the file isn't complete if someone makes a copy while it was being written